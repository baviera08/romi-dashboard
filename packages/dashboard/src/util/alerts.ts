import { createMuiTheme } from '@material-ui/core';
import Swal, { SweetAlertIcon, SweetAlertOptions } from 'sweetalert2';

const theme = createMuiTheme();

export interface VerificationAlertProps {
  title?: string;
  body?: string;
  icon?: SweetAlertIcon;
  confirmButtonText?: string;
  cancelButtonText?: string;
  confirmCallback?: () => void;
  cancelCallback?: () => void;
}

export class Alerts {
  static success = (message?: string) => {
    const swalParams = Alerts.getSuccessMsg(message);
    return Swal.fire(swalParams);
  };

  static error = (message?: string) => {
    const swalParams = Alerts.getErrorMsg(message);
    return Swal.fire(swalParams);
  };

  /**
   * Modal to warn a user about something related to an action.
   */
  static verification = (props: VerificationAlertProps) => {
    const { confirmCallback, cancelCallback } = props;
    const verificationParams = Alerts.getVerificationParams(props);
    Swal.fire(verificationParams).then((modalState) => {
      if (modalState.isConfirmed) {
        !!confirmCallback && confirmCallback();
      } else {
        !!cancelCallback && cancelCallback();
      }
    });
  };

  static getSuccessMsg = (message?: string): SweetAlertOptions => {
    return {
      title: 'Done!',
      text: !!message ? `${message}` : 'Successful Operation',
      icon: 'success',
      timer: 2000,
      heightAuto: false,
    };
  };

  static getErrorMsg = (message?: string): SweetAlertOptions => {
    return {
      title: 'Ups',
      text: !!message ? `${message}` : 'An error has occurred',
      icon: 'error',
      heightAuto: false,
    };
  };

  static getVerificationParams = (props: VerificationAlertProps): SweetAlertOptions => {
    const { title, body, icon, confirmButtonText, cancelButtonText } = props;
    const alertTitle = title || 'Are you sure you want to continue?';
    const alertText = body || 'Once you accept this there is no turning back.';
    const alertIcon = icon || 'warning';
    const alertConfirmButtonText = confirmButtonText || `Yes, I'm sure.`;
    const alertCancelButtonText = cancelButtonText || 'No';

    return {
      title: alertTitle,
      text: alertText,
      icon: alertIcon,
      cancelButtonColor: theme.palette.primary.main,
      confirmButtonColor: theme.palette.error.main,
      showCancelButton: true,
      heightAuto: false,
      confirmButtonText: alertConfirmButtonText,
      cancelButtonText: alertCancelButtonText,
    };
  };
}
