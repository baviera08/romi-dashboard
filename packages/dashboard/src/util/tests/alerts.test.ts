import { createMuiTheme } from '@material-ui/core';
import { Alerts } from '../alerts';

const theme = createMuiTheme();

describe('Verification Alert', () => {
  test('Correct generation of default params values', () => {
    const verification = Alerts.getVerificationOptions({});
    expect(verification).toMatchSnapshot();
  });

  test('Correct generation of custom params values', () => {
    const title = 'test';
    const text = 'test';
    const icon = 'error';
    const confirmButtonText = 'test';
    const cancelButtonText = 'test';

    const verification = Alerts.getVerificationOptions({
      title: title,
      body: text,
      icon: icon,
      confirmButtonText: confirmButtonText,
      cancelButtonText: cancelButtonText,
    });
    expect(verification).toMatchSnapshot();
  });
});

describe('Success Alert', () => {
  test('Correct generation of default params values', () => {
    const success = Alerts.getSuccessOptions();
    expect(success).toEqual({
      title: 'Done!',
      text: 'Successful Operation',
      icon: 'success',
      timer: 2000,
      heightAuto: false,
    });
  });

  test('Correct generation of custom params values', () => {
    const success = Alerts.getSuccessOptions('test');
    expect(success.text).toBe('test');
  });
});

describe('Error Alert', () => {
  test('Correct generation of default params values', () => {
    const error = Alerts.getErrorOptions();
    expect(error).toEqual({
      title: 'Ups',
      text: 'An error has occurred',
      icon: 'error',
      heightAuto: false,
    });
  });

  test('Correct generation of custom params values', () => {
    const error = Alerts.getErrorOptions('test');
    expect(error.text).toBe('test');
  });
});
