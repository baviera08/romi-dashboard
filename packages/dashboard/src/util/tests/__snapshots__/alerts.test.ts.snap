// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verification Alert Correct generation of custom params values 1`] = `
Object {
  "cancelButtonColor": "#3f51b5",
  "cancelButtonText": "test",
  "confirmButtonColor": "#f44336",
  "confirmButtonText": "test",
  "heightAuto": false,
  "icon": "error",
  "showCancelButton": true,
  "text": "test",
  "title": "test",
}
`;

exports[`Verification Alert Correct generation of default params values 1`] = `
Object {
  "cancelButtonColor": "#3f51b5",
  "cancelButtonText": "No",
  "confirmButtonColor": "#f44336",
  "confirmButtonText": "Yes, I'm sure.",
  "heightAuto": false,
  "icon": "warning",
  "showCancelButton": true,
  "text": "Once you accept this there is no turning back.",
  "title": "Are you sure you want to continue?",
}
`;
